### PR TITLE
feat: create docs package for solid components

### PR DESCRIPTION
**Pull request by smfonseca**
This PR is intended to create a separate package which will include all documentation related to SDS.
It's part of [#1131](https://github.com/solid-design-system/solid/issues/1131)

### CODE REVIEW

File: packages/docs/vite.config.ts
Comment (mariohamann) [inline packages/docs/vite.config.ts:1]: Can we reuse things from components vite?

**Comment by mariohamann** [comment]
I think you have to integrate the newest changes on main soon. :)

File: packages/docs/src/stories/templates/video-with-description-and-copyright.stories.ts
Comment (mariohamann) [inline packages/docs/src/stories/templates/video-with-description-and-copyright.stories.ts:1]: Relates to all stories: I think you have to integrate the newest changes from main soon. :)

File: packages/components/tsconfig.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
       }
     }
   ]
- },
- "include": ["src/**/*", "vite.config.*ts"],
- "exclude": ["src/_components/**", "custom-elements-manifest.config.ts"]
+  },
+  "include": ["src/**/*", "vite.config.*ts", "../../docs/src/**/*.ts"],
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/components/tsconfig.json]: Updated.

File: packages/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - `docs`
+    - `changelog`
+  - **`tokens`**
+    - `docs`
+    - `changelog`
+- **`components`**
+- **`utilities`**
+- **`styles`**:
+- **`templates`**
+- **`legal`**
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/README.md]: Removed as discussed.

File: packages/docs/package.json [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    "autoprefixer": "^10.4.16",
+    "cem-plugin-vs-code-custom-data-generator": "^1.4.2",
+    "change-case": "^5.4.4",
+    "chromatic": "^11.12.6",
+    "comment-parser": "^1.4.1",
+    "countup.js": "^2.8.0",
+    "cssnano": "^7.0.3",
+    "lit-html": "^3.1.4",
+    "normalize.css": "^8.0.1",
+    "plop": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/package.json]: Removed

File: packages/docs/postcss.config.cjs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  plugins: {
+    'tailwindcss/nesting': {},
+    tailwindcss: {},
+    autoprefixer: {},
+    ...(process.env.NODE_ENV === 'production' ? { cssnano: {} } : {})
+  }
+};
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/postcss.config.cjs]: Agree, it's better like that.

File: packages/docs/.storybook/preview.js
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/.storybook/preview.js:1]: Synced

File: packages/docs/LICENSE.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+# ASSETS
+
+All code contained in this library is licensed under the [MIT License](LICENSE.md). However, please note that this does not extend to the included assets such as fonts, icons, or images.
+
+These assets are subject to their respective licenses and are included for demonstration purposes and use within the confines of this library. They should not be extracted and used separately without explicit permission from their copyright holders. The unauthorized use of these assets outside the context of this library may infringe on the rights of the respective copyright owners.
+
+If you intend to use any assets outside of this library, you must obtain the necessary permissions from the copyright holders or replace them with your own or license-free alternatives. Failure to comply with asset licensing requirements may result in legal consequences.
+
+Please consult the copyright holders or their licensing documents for further information on using these assets.
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/LICENSE.md:19]: Removed from components license.

File: packages/docs/custom-elements-manifest.config.ts
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/custom-elements-manifest.config.ts:1]: Let's discuss this in a call.

File: packages/docs/src/index.css [hunk 6]
```diff
@@ -0,0 +1,38 @@
+:root {
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/src/index.css]: No, it was auto generated by storybook in the initial setup.

File: packages/docs/src/stories/Docs/General/01-Introduction.mdx
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/src/stories/Docs/General/01-Introduction.mdx:1]: Synced

File: packages/docs/src/stories/Docs/General/01-Introduction.mdx [hunk 7]
```diff
@@ -55,4 +55,4 @@ Our emphasis is on creating a solid foundation of reliable, maintainable compone
...
 > - **Features**: Each component/style now has a single page that contains all the information needed to use it. Our documentation pages now focus on describing all the features a component has, and each feature is shown with a small example. Existing stories were moved to a combining screenshot story per component, which was actually their purpose before.
-> - **Templates**: Before our rewrite, we distinguished between "Samples" and "Patterns". This sometimes caused confusion. We decided to combine the two concepts into one and rename them to "Templates", hoping that this would better reflect the understanding of "copy-pastable". They are now listed at the same page level, and additionally at each component used in the template.
+> - **Templates**: Before our rewrite, we distinguished between "Samples" and "Patterns". This sometimes caused confusion. We decided to combine the two concepts into one and rename them to "Templates", hoping that this would better reflect the understanding of "copy-pastable". They are now listed at the same page level, and additionally at each component used in the template.../../../package.json
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/src/stories/Docs/General/01-Introduction.mdx]: Hmmm, not sure why this ended up here. 
Removed.

File: packages/docs/src/stories/Docs/General/03-Installation.mdx [hunk 8]
```diff
@@ -151,3 +151,4 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 3. If you want to receive even breaking changes by using the latest version, you can use the wildcard major version selector: <Markdown>{`\`x.x.x\``}</Markdown>
 
 > Important: In production environments you should always use a fixed version, as otherwise you could receive breaking changes without noticing. Using a more flexible version is only recommended for development environments and at own risk.
+> ../../../package.json
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/docs/src/stories/Docs/General/03-Installation.mdx]: Agree, it's better like that.

File: packages/docs/vite.config.ts
Comment (smfonseca) [reply packages/docs/vite.config.ts:1]: Let's discuss this in a call.

File: packages/docs/src/stories/templates/video-with-description-and-copyright.stories.ts
Comment (smfonseca) [reply packages/docs/src/stories/templates/video-with-description-and-copyright.stories.ts:1]: I will do this once we have everything else done and this is the last step.

**Review by mariohamann — APPROVED** [review]
Beautiful.

**Pull request merged**